api/vulnerability_exceptions.go

```diff
@@ -0,0 +1,453 @@
+		sev := convertVulnerabilityExceptionSeverityInt(i)
+		if sev != VulnerabilityExceptionSeverityUnknown {
+			res = append(res, sev)
+		}
+	}
+	return res
+}
+
+func convertVulnerabilityExceptionSeverity(sev string) vulnerabilityExceptionSeverity {
+	switch strings.ToLower(sev) {
+	case "critical":
+		return VulnerabilityExceptionSeverityCritical
+	case "high":
+		return VulnerabilityExceptionSeverityHigh
+	case "medium":
+		return VulnerabilityExceptionSeverityMedium
+	case "low":
+		return VulnerabilityExceptionSeverityLow
+	case "info":
+		return VulnerabilityExceptionSeverityInfo
+	default:
+		return VulnerabilityExceptionSeverityUnknown
+	}
+}
+
+func convertVulnerabilityExceptionSeverityInt(sev int) vulnerabilityExceptionSeverity {
+	switch sev {
+	case 1:
+		return VulnerabilityExceptionSeverityCritical
+	case 2:
+		return VulnerabilityExceptionSeverityHigh
+	case 3:
+		return VulnerabilityExceptionSeverityMedium
+	case 4:
+		return VulnerabilityExceptionSeverityLow
+	case 5:
+		return VulnerabilityExceptionSeverityInfo
+	default:
+		return VulnerabilityExceptionSeverityUnknown
+	}
+}
+
+const (
+	VulnerabilityExceptionSeverityCritical vulnerabilityExceptionSeverity = 1
+	VulnerabilityExceptionSeverityHigh     vulnerabilityExceptionSeverity = 2
+	VulnerabilityExceptionSeverityMedium   vulnerabilityExceptionSeverity = 3
+	VulnerabilityExceptionSeverityLow      vulnerabilityExceptionSeverity = 4
+	VulnerabilityExceptionSeverityInfo     vulnerabilityExceptionSeverity = 5
+	VulnerabilityExceptionSeverityUnknown  vulnerabilityExceptionSeverity = 0
+)
+
+// NewVulnerabilityException returns an instance of the VulnerabilityException struct
+//
+// Basic usage: Initialize a new VulnerabilityException struct, then
+//              use the new instance to do CRUD operations
+//
+//   client, err := api.NewClient("account")
+//   if err != nil {
+//     return err
+//   }
+//
+//   exception := api.VulnerabilityExceptionConfig{
+//       Type:            api.VulnerabilityExceptionTypeHost,
+//       Description:     "This is a vuln exception",
+//       ExceptionReason: api.VulnerabilityExceptionReasonCompensatingControls,
+//       Severities:      api.VulnerabilityExceptionSeverities{api.VulnerabilityExceptionSeverityCritical},
+//       Fixable: 		  true,
+//       ResourceScope:   api.VulnerabilityExceptionContainerResourceScope{
+//          ImageID:    []string{""},
+//          ImageTag:   []string{""},
+//          Registry:   []string{""},
+//          Repository: []string{""},
+//          Namespace:  []string{""},
+//       },
+//       ExpiryTime: time.Now().AddDate(0, 1, 0),
+//   }
+//
+//	 vulnerabilityException := api.NewVulnerabilityException("vulnerabilityException", exception)
+//
+//   client.V2.VulnerabilityExceptions.Create(vulnerabilityException)
+//
+func NewVulnerabilityException(name string, exception VulnerabilityExceptionConfig) VulnerabilityException {
+	packages := AggregatePackages(exception.Package)
```

Comment:
Is `AggregatePackages` public on purpose? Can we prefix the function to be something like `VulnerabilityExceptionAggregatePackages()` just because there could be other "aggregation of packages" that is not for vulnerability exceptions

Reply:
I will change it to private.